n3fit/src/n3fit/scripts/vp_setupfit.py

```diff
@@ -157,6 +157,9 @@ def from_yaml(cls, o, *args, **kwargs):
         if file_content.get('theorycovmatconfig') is not None:
             SETUPFIT_FIXED_CONFIG['actions_'].append(
                 'datacuts::theory::theorycovmatconfig nnfit_theory_covmat')
+        if file_content.get('fiatlux') is not None:
+            SETUPFIT_FIXED_CONFIG['actions_'].append(
+                'fiatlux check_luxset')
```

Comment:
You can just pass `luxset` to performfit without adding the `check_luxset`. If you have a look at what `parse_pdf` does you'll see that it already loads the pdf
https://github.com/NNPDF/nnpdf/blob/fcb766bf84e4a72c0aedcf955265f13f991f2d6e/validphys2/src/validphys/config.py#L148

To be honest, I'm not really sure my proposal is the cleanest way to do things. Particularly because luxset is used only in n3fit and not inside vp, and you already pass the fiatlux dictionary (so the information would be double), but for now it's the best I can come up with...